### PR TITLE
[Feature][Connector-V2] new connecotor of Elasticsearch sink(#2326)

### PR DESCRIPTION
**Pull request by iture123**
## Purpose of this pull request
This pull request  add new connecotor of Elasticsearch sink.(#2326)

## Check list

* [ √] Code changed are covered with tests, or it does not need tests for reason:
* [ √ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [√ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

File: seatunnel-connectors-v2/connector-elasticsearch/pom.xml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    <dependencies>
+        <dependency>
+            <groupId>org.apache.seatunnel</groupId>
+            <artifactId>seatunnel-api</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.elasticsearch.client</groupId>
+            <artifactId>elasticsearch-rest-client</artifactId>
+            <version>8.1.3</version>
```
Comment (CalvinKirs) [inline seatunnel-connectors-v2/connector-elasticsearch/pom.xml]: The version is best managed uniformly in the `root pom`.(DependencyManagent)

File: seatunnel-connectors-v2/connector-elasticsearch/src/java/org/apache/seatunnel/connectors/seatunnel/elasticsearch/client/EsRestClient.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        }
+        return esRestClient;
+    }
+
+    public BulkResponse bulk(String requestBody) {
+        Request request = new Request("POST", "_bulk");
+        request.setJsonEntity(requestBody);
+        try {
+            Response response = restClient.performRequest(request);
+            if (response.getStatusLine().getStatusCode() == 200) {
```
Comment (CalvinKirs) [inline seatunnel-connectors-v2/connector-elasticsearch/src/java/org/apache/seatunnel/connectors/seatunnel/elasticsearch/client/EsRestClient.java]: It's better use` HttpStatus` judge.

Comment (CalvinKirs) [reply seatunnel-connectors-v2/connector-elasticsearch/src/java/org/apache/seatunnel/connectors/seatunnel/elasticsearch/client/EsRestClient.java]: And maybe be NPE?

Comment (iture123) [reply seatunnel-connectors-v2/connector-elasticsearch/src/java/org/apache/seatunnel/connectors/seatunnel/elasticsearch/client/EsRestClient.java]: I have revised  it according to your suggestion.

File: seatunnel-connectors-v2/connector-elasticsearch/pom.xml (hunk 1 above)
Comment (iture123) [reply seatunnel-connectors-v2/connector-elasticsearch/pom.xml]: Yes,I have optimized it.

**Comment by CalvinKirs** [comment]
could you submit this PR to dev branch(not api-draft)

**Comment by CalvinKirs** [comment]
@Hisoka-X hi,PTAL,thx

**Review by Hisoka-X — DISMISSED** [review]

**Comment by Hisoka-X** [comment]
Hi please resolve dependency check error

**Review by CalvinKirs — APPROVED** [review]

**Pull request merged**